components/date-picker/RangePicker.tsx

```diff
@@ -62,13 +64,31 @@ function fixLocale(value: RangePickerValue | undefined, localeCode: string) {
   }
 }
 
-export default class RangePicker extends React.Component<any, RangePickerState> {
+class RangePicker extends React.Component<any, RangePickerState> {
   static defaultProps = {
     prefixCls: 'ant-calendar',
     allowClear: true,
     showToday: false,
   };
 
+  static getDerivedStateFromProps(nextProps: any, prevState: any) {
+    let state = null;
+    if (('value' in nextProps) && !isEqual(nextProps.value,prevState.value)) {
```

Comment:
Why need to compare `nextProps.value` and `prevState.value`?

Reply:
@yesmeck so that it only change state when the value of 'value' changes, otherwise it'll reset value of 'value' every time the parent rerenders as @whtsky commented

Reply:
Is it ok to reset value of `value` on every time the parent rerenders? Like we did in `componentWillReceiveProps`?

Reply:
yup we did reset it every time in componentWillReceiveProps so I suppose its okay to do so ?
Ok I'll change it